packages/core/src/browser/frontend-idle-status.ts

```diff
@@ -0,0 +1,136 @@
+
+/**
+ * Service providing information whether the application is idle or not.
+ */
+export const FrontendIdleService = Symbol('ApplicationIdleService');
+export interface FrontendIdleService {
+
+    /**
+     * The current idle state of the application. `true` is idle, `false` is active.
+     */
+    readonly idle: boolean;
+
+    /**
+     * Clients could listen on the idle state of the frontend application.
+     */
+    readonly onStatusChange: Event<boolean>;
+
+}
+
+@injectable()
+export class IdleStatusOptions {
+
+    /**
+     * The default idle options. The application goes into idle state after 15 seconds.
+     */
+    static DEFAULT: IdleStatusOptions = {
+        idleAfter: 15000
+    };
+
+    /**
+     * Time (in milliseconds) after the frontend application goes to idle state.
+     */
+    readonly idleAfter: number;
+
+}
+
+@injectable()
+export class DefaultFrontendIdleService implements FrontendIdleService, FrontendApplicationContribution {
+
+    protected readonly emitter: Emitter<boolean>;
+    protected readonly handler: any;
+
+    private _idle = false;
+    private timer: number | undefined;
+
+    constructor(
+        @inject(IdleStatusOptions) @optional() protected readonly options: IdleStatusOptions = IdleStatusOptions.DEFAULT,
+        @inject(ILogger) protected readonly logger: ILogger
+    ) {
+        this.handler = this.resetTimer.bind(this);
+        this.emitter = new Emitter<boolean>();
+        this.fireStateChange(this._idle);
+    }
+
+    onStart() {
+        this.setup();
+    }
+
+    onStop() {
+        document.removeEventListener('mousemove', this.handler);
+        document.removeEventListener('mousedown', this.handler);
+        document.removeEventListener('keypress', this.handler);
+        document.removeEventListener('DOMMouseScroll', this.handler);
+        document.removeEventListener('mousewheel', this.handler);
+        document.removeEventListener('touchmove', this.handler);
+        document.removeEventListener('MSPointerMove', this.handler);
+        this.clearTimer();
+    }
+
+    get onStatusChange() {
+        return this.emitter.event;
+    }
+
+    get idle() {
+        return this._idle;
+    }
+
+    protected setup() {
+        document.addEventListener('mousemove', this.handler, false);
+        document.addEventListener('mousedown', this.handler, false);
+        document.addEventListener('keypress', this.handler, false);
+        document.addEventListener('DOMMouseScroll', this.handler, false);
+        document.addEventListener('mousewheel', this.handler, false);
+        document.addEventListener('touchmove', this.handler, false);
+        document.addEventListener('MSPointerMove', this.handler, false);
+        this.startTimer();
+    }
+
+    protected resetTimer() {
+        this.clearTimer();
+        this.goActive();
+    }
+
+    protected clearTimer() {
+        if (this.timer !== undefined) {
+            window.clearTimeout(this.timer);
```

Comment:
`window.clearInterval`

Reply:
No, that is correct. The counterpart is the broken one. Thanks!